import "common-proto.proto";

package strongstore.proto;

message GetMessage {
    repeated string keys = 1;
    optional uint64 timestamp = 2;
}

message PrepareMessage {
    required TransactionMessage txn = 1;
}

message CommitMessage {
    required uint64 timestamp = 1;
    optional TransactionMessage txn = 2;
}

message AbortMessage {
}

message Request {
     enum Operation {
          GET = 1;
          PREPARE = 2;
          COMMIT = 3;
          ABORT = 4;
     }	
     required Operation op = 1;
     required uint64 txnid = 2;
     optional GetMessage get = 3;
     optional PrepareMessage prepare = 4;
     optional CommitMessage commit = 5;
     optional AbortMessage abort = 6;
}

message Reply {
     required int32 status = 1;
     repeated ReadReply replies = 2;
     optional uint64 timestamp = 3;
}
